File: app/src/main/java/com/backupbro/model/UserViewModel.java
package com.backupbro.model;

import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.ViewModel;
import com.backupbro.repository.UserRepository;
import javax.inject.Inject;
import io.reactivex.Observable;

public class UserViewModel extends ViewModel {
    private LiveData<User> user;
    private UserRepository userRepository;

    @Inject
    public UserViewModel(UserRepository userRepository) {
        this.userRepository = userRepository;
    }

//    public void login(String email, String password) {
//        this.user = this.userRepository.login(email, password);
//    }
//
//    public LiveData<User> getUser() {
//        return user;
//    }

//
}

File: app/src/main/java/com/backupbro/model/UserViewModel.kt
package com.backupbro.model

import android.arch.lifecycle.ViewModel
import com.backupbro.repository.UserRepository
import io.reactivex.Observable
import javax.inject.Inject

class UserViewModel @Inject constructor(
        val userRepository: UserRepository
    ): ViewModel() {

    lateinit var user : Observable<User>

    fun login(email: String, password: String) : Observable<User> {
        this.user = userRepository.login(email, password)
        return this.user
    }

    fun register(email: String, password: String, username: String) : Observable<User> {
        this.user = userRepository.register(email, password, username)
        return this.user
    }


}